app/src/androidTest/java/com/losandroides/learn/presentation/MainActivityTest.kt

```diff
@@ -0,0 +1,87 @@
+package com.losandroides.learn.presentation
+
+import androidx.compose.ui.test.assertCountEquals
+import androidx.compose.ui.test.assertIsDisplayed
+import androidx.compose.ui.test.assertTextEquals
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onChildAt
+import androidx.compose.ui.test.onChildren
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.compose.ui.test.onNodeWithText
+import com.losandroides.learn.domain.model.Item
+import org.junit.Rule
+import org.junit.Test
+
+class MainActivityTest {
+
+    @get:Rule
+    val composeTestRule = createComposeRule()
+
+    private fun setUp(
+        viewModelState: MainViewModel.ViewState
+    ) {
+        composeTestRule.setContent {
+            ItemListScreen(viewModelState = viewModelState)
+        }
+    }
+
+    @Test
+    fun givenLoadingStateThenLoaderIsDisplayed() {
+        setUp(MainViewModel.ViewState.Loading)
+
+        composeTestRule.onNodeWithTag("loader").assertIsDisplayed()
```

Comment:
Just asking: Would make sense to have the `"loader"` as a public shared constant somehow?

Reply:
You mean the constant itself or the `@Composable`? 
If you mean the constant, we can add "loader" as a `const val LOADER_TEST_TAG="loader"` but in the test, I'd use the "hardcoded" text anyways, I'm not fan of using production constants in test, since then it's not a real "test".
If you mean `@Composable` yes, it would make sense in case we want to add a common loader for all the app.

Reply:
I meant the `loader` string.
I understand your point about "production constants in test".
And I think it makes sense.
Thanks for clarifying! :hugs:

Reply:
I always try to use first content descriptions, keeps your app more accessible this way and you can use them in the tests with `.onNodeWithContentDescription()`